### PR TITLE
Add some parameters to repositories

### LINKED ISSUE #581: katello_repository: Allow setting of upstream username and password for authenticated upstream repos
##### SUMMARY
If I have an upstream repo that requires authentication, I ned to supply a username and password with the katello_repository module. There is no way to do so currently.

##### ISSUE TYPE
 - Feature Idea

### PR DESCRIPTION
**Pull request by mdellweg**
fixes #581

### CODE REVIEW

**Comment by mdellweg** [comment]
cc @jwbernin

File: plugins/modules/katello_repository.py [hunk 1]
```diff
@@ -88,10 +83,41 @@
...
+      - username to access upstream repository
+    type: str
+  upstream_password:
+    description:
+      - password to access upstream repository
+    type: str
+  docker_upstream_name:
+    description:
+      - name of the upstream docker repository
+      - only available for I(content_type)==I(docker)
```
Comment (evgeni) [inline plugins/modules/katello_repository.py]: the [examples](https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#linking-within-module-documentation) use `I(content_type=docker)`
```suggestion
      - only available for I(content_type=docker)
```

File: plugins/modules/katello_repository.py [hunk 2]
```diff
@@ -88,10 +83,41 @@
...
+    type: str
+  docker_tags_whitelist:
+    description:
+      - list of tags to sync for Container Image repository
+      - only available for I(content_type)==I(docker)
```
Comment (evgeni) [inline plugins/modules/katello_repository.py]:
```suggestion
      - only available for I(content_type=docker)
```

File: plugins/modules/katello_repository.py [hunk 3]
```diff
@@ -88,10 +83,41 @@
...
+    type: list
+  deb_releases:
+    description:
+      - comma separated list of releases to be synched from deb-archive
```
Comment (evgeni) [inline plugins/modules/katello_repository.py]:
```suggestion
      - comma separated list of releases to be synced from deb-archive
      - only available for I(content_type=deb)
```

File: plugins/modules/katello_repository.py [hunk 4]
```diff
@@ -88,10 +83,41 @@
...
+    type: str
+  deb_components:
+    description:
+      - comma separated list of repo components to be synched from deb-archive
```
Comment (evgeni) [inline plugins/modules/katello_repository.py]:
```suggestion
      - comma separated list of repo components to be synced from deb-archive
      - only available for I(content_type=deb)
```

File: plugins/modules/katello_repository.py [hunk 5]
```diff
@@ -88,10 +83,41 @@
...
+    type: str
+  deb_architectures:
+    description:
+      - comma separated list of architectures to be synched from deb-archive
```
Comment (evgeni) [inline plugins/modules/katello_repository.py]:
```suggestion
      - comma separated list of architectures to be synced from deb-archive
      - only available for I(content_type=deb)
```

**Comment by mdellweg** [comment]
@evgeni Yes, that is how katello hands it over to pulp.
And it might become a more complex language that just a comma separated whitelist some time in the future.

**Pull request merged**